src/commands/app/upgrade.tsx

```diff
@@ -114,7 +115,19 @@ export class UpgradeApp extends ExecRenderBaseCommand<typeof UpgradeApp, void> {
           currentApp.id,
           currentAppVersion.id,
         )) as AppAppVersion;
-    } else if (this.flags["target-version"]) {
+    } else if (
+      this.flags["target-version"] &&
+      validateUuid(this.flags["target-version"])
+    ) {
+      targetAppVersion = await getAppVersionFromUuid(
+        this.apiClient,
+        currentApp.id,
+        this.flags["target-version"],
+      );
+    } else if (
+      this.flags["target-version"] &&
+      !validateUuid(this.flags["target-version"])
+    ) {
```

Comment:
These `AND`ed conditions could be deconstructed to (maybe?) more readable nested conditions:

```
else if (this.flags["target-version"]) {
  if (validateUuid(this.flags["target-version"])) {
    // ...
  } else {
    // ...
  }
} // ...
```

Also, the way how `targetAppVersion` is getting more and more complex, with more and more branches in this condition. I would suggest extracting the entire logic of populating `targetAppVersion` into a function.